packages/amplication-server/src/core/prismaSchemaUtils/prismaSchemaUtils.service.ts

```diff
@@ -377,14 +379,21 @@ export class PrismaSchemaUtilsService {
       const formattedModelName = formatModelName(model.name);
 
       if (formattedModelName !== model.name) {
+        const isFormattedModelNameAlreadyTaken = models.some(
+          (model) => model.name === formattedModelName
+        );
```

Comment:
This test only checks against the original model names, but you still may have a duplicate name with a new formatted name of a previously prepared model